src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -381,6 +376,7 @@ public MqttClient(DeviceConfiguration deviceConfiguration, Spool spool, boolean
 
         this.deviceConfiguration = deviceConfiguration;
         mqttTopics = this.deviceConfiguration.getMQTTNamespace();
+        this.topicValidator = new IotCoreTopicValidator(deviceConfiguration);
```

Comment:
I'd probably simplify and just pass the mqtt version at the lookup time

Reply:
done